src/query-builder/UpdateQueryBuilder.ts

```diff
@@ -129,7 +134,10 @@ export class UpdateQueryBuilder<Entity> extends QueryBuilder<Entity> implements
      * Values needs to be updated.
      */
     set(values: ObjectLiteral): this {
-        this.expressionMap.valuesSet = values;
+        const copiedValues = {...values};
```

Comment:
I won't do it this way. `values` can be an entity and it used as an entity down the road so we'll have the issues. Also if its an entity it may have properties other then column values so we need to check only column values.